hazelcast/include/hazelcast/client/protocol/ClientMessage.h

```diff
@@ -1284,6 +1292,44 @@ class HAZELCAST_API ClientMessage
         add_end_frame(is_final);
     }
 
+    void set(const serialization::pimpl::field_descriptor& descriptor,
+             const std::string& field_name,
+             bool is_final = false)
+    {
+        add_begin_frame();
+
+        set(frame_header_type{ SIZE_OF_FRAME_LENGTH_AND_FLAGS + INT32_SIZE,
+                               DEFAULT_FLAGS });
+        set(int32_t(descriptor.kind));
+        set(field_name);
+
+        add_end_frame(is_final);
+    }
+
+    void set(const serialization::pimpl::schema& s, bool is_final = false)
+    {
+        add_begin_frame();
+
+        set(s.type_name());
+
+        { // Fields list
+            add_begin_frame();
```

Comment:
Do we need to add a begin frame here? In field descriptor codec it start with add_begin_frame. In Java client, there is not a frame begin before for loop.